services/components/lifecycle/bot_logger_test.go

```diff
@@ -0,0 +1,38 @@
+package lifecycle
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/forta-network/forta-core-go/clients/agentlogs"
+	"github.com/forta-network/forta-core-go/security"
+	mock_clients "github.com/forta-network/forta-node/clients/mocks"
+	mock_containers "github.com/forta-network/forta-node/services/components/containers/mocks"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+)
+
+func sendLogsMock(agents agentlogs.Agents, authToken string) error {
+	return nil
+}
+
+func TestNewBotLogger(t *testing.T) {
+	ctrl := gomock.NewController(t)
+
+	r := require.New(t)
+
+	botClient := mock_containers.NewMockBotClient(ctrl)
+	dockerClient := mock_clients.NewMockDockerClient(ctrl)
+
+	dir := t.TempDir()
+	ks := keystore.NewKeyStore(dir, keystore.StandardScryptN, keystore.StandardScryptP)
+
+	_, err := ks.NewAccount("Forta123")
+	r.NoError(err)
+
+	key, err := security.LoadKeyWithPassphrase(dir, "Forta123")
+	r.NoError(err)
+
+	botLogger := NewBotLogger(botClient, dockerClient, key, sendLogsMock)
+	r.NotNil(botLogger)
```

Comment:
After this line, I would call `botLogger.SendBotLogs(context.Background())` and expect some other calls. This requires setting up some expected calls on `botClient` and `dockerClient` first, by using `.EXPECT()` on each.

And instead of defining a function like `sendLogsMock`, I would pass an anonymous function to `NewBotLogger` (like `func() { ... }`) and do `r.Equal` kind of assertions in that.

Reply:
Sounds good :) I wanted to add another test for testing SendLogs, and use this as the Setup